File: docs/code-examples/line_segments3d_simple_v2.cpp
// Log a simple set of line segments.

#include <rerun.hpp>

namespace rr = rerun;

int main() {
    auto rr_stream = rr::RecordingStream("line_segments3d");
    rr_stream.connect("127.0.0.1:9876").throw_on_failure();

    std::vector<rr::datatypes::Vec3D> points = {
        {0.f, 0.f, 0.f},
        {0.f, 0.f, 1.f},
        {1.f, 0.f, 0.f},
        {1.f, 0.f, 1.f},
        {1.f, 1.f, 0.f},
        {1.f, 1.f, 1.f},
        {0.f, 1.f, 0.f},
        {0.f, 1.f, 1.f},
    };
    rr_stream.log("segments", rr::archetypes::LineStrips3D(points));
}

File: docs/code-examples/line_segments3d_simple_v2.py
"""Log a simple set of line segments."""
import numpy as np
import rerun as rr
import rerun.experimental as rr2

rr.init("rerun-example-line_segments3d", spawn=True)

rr2.log(
    "segments",
    rr2.LineStrips3D(
        np.array(
            [
                [[0, 0, 0], [0, 0, 1]],
                [[1, 0, 0], [1, 0, 1]],
                [[1, 1, 0], [1, 1, 1]],
                [[0, 1, 0], [0, 1, 1]],
            ],
        )
    ),
)
